sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -896,13 +933,12 @@ private SqlTestFramework(Builder builder)
         // test pulls in a module, then pull in that module, even though we are
         // not the Druid node to which the module is scoped.
         .ignoreLoadScopes();
-    List<Module> overrideModules = new ArrayList<>(builder.overrideModules);
-    overrideModules.add(new LookylooModule());
-    overrideModules.add(new SqlAggregationModule());
-    overrideModules.add(new SegmentWranglerModule());
-    overrideModules.add(new ExpressionModule());
 
+    ArrayList<Module> overrideModules = new ArrayList<>(builder.overrideModules);
+
+    injectorBuilder.add(componentSupplier.getCoreModule());
     overrideModules.add(testSetupModule());
+    overrideModules.add(componentSupplier.getOverrideModule());
```

Comment:
The base framework should not add anything to override modules.  Override modules exist to allow the *tests* to override things that the framework setup.  If the framework adds stuff to overrides here, then the tests cannot override those things, causing problems when tests need to change behaviors.  The override modules need to be wholly owned and controlled by the test setup.  The fact that these modules are being added as overrides points to the framework not being setup properly.  If it is running into competing bindings, it needs to resolve them by either overriding the competing module when it's added *or* by applying these overrides in a way that allows for further override by the tests.

Reply:
> The override modules need to be wholly owned and controlled by the test setup

The test could alter the `componentSupplier` and that should be accessible by it - this line should essentially just plug those overrides in.

If the test uses a custom componentsupplier; that could customize `getOverrideModule` as well; if it has conflict with the ComponentSupplier its being derived from with by `extends` ; it may use `Modules.override` to override those - but I'll try to avoid the need for that; as that's just ugly....

Reply:
based on offline discussion: the problem here was that `testSetupModule()` was fully wired in - which could become problematic

I've made the `TestSetupModule` added by the `componentSupplier` in the `OverrideModule` ; so it will be easier to override even these things more easily

I've also reorganized some parts of `TestSetupModule`  into a separate module - for which later there will be a different implementation.